`error: ‘pthread_mutex_t’ has no member named ‘abi’` in refc with `reset`/`{.exportc.}`

### Nim Version

```
Nim Compiler Version 2.2.4 [Linux: amd64]
Compiled at 2025-10-06
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: f7145dd26efeeeb6eeae6fff649db244d81b212d
active boot switches: -d:release
```

```
Nim Compiler Version 2.2.5 [Linux: amd64]
Compiled at 2025-10-06
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: cfe163795f3a8ac8ecf11dc0ab788a3758b2a32e
active boot switches: -d:release
```

```
Nim Compiler Version 2.3.1 [Linux: amd64]
Compiled at 2025-10-06
Copyright (c) 2006-2025 by Andreas Rumpf

git hash: 483389d3999e243d7c61647c55187b0578dc57ee
active boot switches: -d:release
```

```
gcc (Debian 15.2.0-4) 15.2.0
Copyright (C) 2025 Free Software Foundation, Inc.
```

### Description

```nim
import std/locks

type
  S = object
    r: proc()

  B = object
    d: Lock
    w: S

proc v(x: ptr B) {.exportc.} = reset(x[])
```

### Current Output

```text
CC: system/exceptions.nim
CC: std/private/digitsutils.nim
CC: system/dollars.nim
CC: std/typedthreads.nim
CC: sharedlist.nim
CC: system.nim
CC: types.nim
/tmp/tmp.m32umU53cV/@psystem.nim.c: In function ‘reset__types_u10’:
/tmp/tmp.m32umU53cV/@psystem.nim.c:5491:20: error: ‘pthread_mutex_t’ has no member named ‘abi’
 5491 |         (*obj_p0).d.abi[T1_] = 0;
```

### Expected Output

```text
No C compilation error
```

### Known Workarounds

_No response_

### Additional Information

_No response_

--- a/compiler/ccgreset.nim
+++ b/compiler/ccgreset.nim
@@ -67,7 +67,14 @@ proc specializeResetT(p: BProc, accessor: Rope, typ: PType) =
     var x = typ.baseClass
     if x != nil: x = x.skipTypes(skipPtrs)
     specializeResetT(p, accessor.parentObj(p.module), x)
-    if typ.n != nil: specializeResetN(p, accessor, typ.n, typ)
+    if typ.n != nil:
+      if typ.sym != nil and sfImportc in typ.sym.flags:
+        # imported C struct, nimZeroMem
+        p.s(cpsStmts).addCallStmt(cgsymValue(p.module, "nimZeroMem"),
+          cCast(ptrType(CPointer), cAddr(accessor)),
+          cSizeof(getTypeDesc(p.module, typ)))
+      else:
+        specializeResetN(p, accessor, typ.n, typ)
   of tyTuple:
     let typ = getUniqueType(typ)
     for i, a in typ.ikids:

--- a/tests/stdlib/tlocks.nim
+++ b/tests/stdlib/tlocks.nim
@@ -9,3 +9,30 @@ import std/assertions
 
 var m = createMyType[int]()
 doAssert m.use() == 3
+
+
+import std/locks
+
+type
+  S = object
+    r: proc()
+
+  B = object
+    d: Lock
+    w: S
+
+proc v(x: ptr B) {.exportc.} = reset(x[])
+
+type
+  Test = object
+    path: string # Removing this makes both cases work.
+    lock: Lock
+
+# A: This is not fine.
+var a = Test()
+
+proc main(): void =
+  # B: This is fine.
+  var b = Test()
+
+main()

